tests/python/pytest_input_generators.py

```diff
@@ -1478,3 +1478,31 @@ def vector_at_error_generator(
         yield SampleInput(
             make_arg(error_case["tensor_shape"]), index=error_case["index"]
         ), error_type, error_msg
+
+
+def matmul_input_generator(
+    op: OpInfo, dtype: torch.dtype, requires_grad: bool = False, **kwargs
+):
+    make_arg = partial(
+        make_tensor,
+        dtype=torch.float16,
+        device="cuda",
+        low=None,
+        high=None,
+        requires_grad=requires_grad,
+    )
+
+    def multiply_range(maximum, step):
+        assert maximum % step == 0
+        num_steps = int(math.log(maximum, step))
+        return tuple(
+            map(pow, itertools.repeat(step, num_steps), range(1, num_steps + 1))
+        )
+
+    # Ranges of tensor sizes: 8, 64, 512, 4096, 32768
+    # Use a Cartesian product to create a wide range of matrix shapes
```

Comment:
Does this use the same `FusionDefinition` to run all the cases or will it be compiling 125 separate kernels?

Reply:
It only compiles once (twice - once for fp16 and again for bf16).

Reply:
We are creating a new `FusionDefinition` for each case, but they should map to the same `FusionExecutorCache` in the python frontend `FusionCache`.

https://github.com/NVIDIA/Fuser/blob/main/tests/python/pytest_ops.py#L47-L48

The total number of cuda kernels would depend on NvFuser's heuristics. e.g., Some kernels could be reused for some matrix shapes.

Reply:
Thanks! My answer was based on debugging via prints. :)